core/src/main/java/io/parsingdata/metal/Shorthand.java

```diff
@@ -152,6 +152,10 @@ private Shorthand() {}
     public static Token until(final String name, final ValueExpression initialSize, final Token terminator) { return until(name, initialSize, null, terminator, null); }
     public static Token until(final String name, final Token terminator, final Encoding encoding) { return until(name, null, terminator, encoding); }
     public static Token until(final String name, final Token terminator) { return until(name, terminator, null); }
+    public static Token when(final String name, final Token token, final Expression predicate, final Encoding encoding) { return cho(name, encoding, pre(def(EMPTY_NAME, 0), not(predicate)), token); }
```

Comment:
Just checking: this is exactly how we implemented `iff`?

Reply:
Well, @ccreeten provided me with this idea and I think he implemented the original `iff`...

Reply:
I'm not familiar with the details of the `iff` implementation, I'm just checking if this exactly is the same :wink: